Face/FindSimilar/Program.cs

```diff
@@ -0,0 +1,258 @@
+
+                Console.WriteLine($"{detectedFaces.Count} faces detected from image `{SourceImageFileName}`.");
+                if (detectedFaces[0].FaceId == null)
+                {
+                    Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                    return;
+                }
+
+                // Find similar example for find similar faces from targetFaceIds with similar confidence.
+                IList<SimilarFace> similarResults = await Client.Face.FindSimilarAsync(detectedFaces[0].FaceId.Value, null, null, targetFaceIds);
+                if (similarResults.Count == 0)
+                {
+                    Console.WriteLine($"Not found the same with `{detectedFaces[0].FaceId}` from similar result.{Environment.NewLine}");
+                }
+
+                foreach (var similarResult in similarResults)
+                {
+                    Console.WriteLine(
+                        similarResult.Confidence > 0.5
+                            ? $"Faces from {detectedFaces[0].FaceId} & {similarResult.FaceId} are of the same (Positive) person, similarity confidence: {similarResult.Confidence}."
+                            : $"Faces from {detectedFaces[0].FaceId} & {similarResult.FaceId} are of different (Negative) persons, similarity confidence: {similarResult.Confidence}.");
+                }
+            }
+        }
+
+        public static async Task FindSimilar_in_FaceList()
+        {
+            // Create a face list.
+            string faceListId = Guid.NewGuid().ToString();
+            Console.WriteLine($"Create FaceList {faceListId} {Environment.NewLine}");
+            await Client.FaceList.CreateAsync(faceListId, faceListId, "face list for sample");
+
+            foreach (var targetImageFileName in TargetImageFileNames)
+            {
+                // Read image file.
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Add face to face list.
+                    Console.WriteLine($"Add faces to face list from image `{targetImageFileName}`.");
+                    var faces = await Client.FaceList.AddFaceFromStreamAsync(faceListId, stream, targetImageFileName);
+                    if (faces == null)
+                    {
+                        Console.WriteLine($"[Error] No persisted face from image `{targetImageFileName}`.");
+                        return;
+                    }
+                }
+            }
+
+            // Get persisted faces.
+            List<PersistedFace> persistedFaces = (await Client.FaceList.GetAsync(faceListId)).PersistedFaces as List<PersistedFace>;
+            if (persistedFaces == null)
+            {
+                Console.WriteLine($"[Error] Persisted face not found in face list \"{faceListId}'.");
+                return;
+            }
```

Comment:
// Get persisted faces from the face list.
            List<PersistedFace> persistedFaces = (await Client.FaceList.GetAsync(faceListId)).PersistedFaces.ToList();
            if (persistedFaces.Count == 0)
            {
                Console.WriteLine($"[Error] No persisted face in face list \"{faceListId}'.");
                return;
            }


Same for large face list.

Reply:
done